### PR TITLE
Publicize Third Party Authentication methods

### PR DESCRIPTION
**Pull request by grantland**
Publicize + JavaDoc. Required for open sourcing ParseFacebookUtils and ParseTwitterUtils.

### CODE REVIEW

File: Parse/src/main/java/com/parse/ParseAuthenticationProvider.java [hunk 1]
```diff
@@ -13,36 +13,29 @@
...
-   */
-  Task<Map<String, String>> authenticateAsync();
-
-  /**
-   * Deauthenticates (logs out) the user associated with this provider. This call may block.
-   *
-   * @return A {@link Task} that resolves when deauthentication is complete.
+   * @return A {@link Task} that resolves when deauthentication completes.
    */
   Task<Void> deauthenticateAsync();
```
Comment (hallucinogen) [inline Parse/src/main/java/com/parse/ParseAuthenticationProvider.java]: This should be `deauthenticateInBackground`?

Comment (grantland) [inline Parse/src/main/java/com/parse/ParseAuthenticationProvider.java]: good catch!

**Comment by grantland** [comment]
@hallucinogen updated

**Comment by hallucinogen** [comment]
Looking good! @nlutsenko should double-check tho since this is a public API.

**Comment by nlutsenko** [comment]
LGTM

**Pull request merged**